CEGameApp.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CEGameApp.API.Data;
+using CEGameApp.API.Dtos;
+using CEGameApp.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace CEGameApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthRepository _repo;
+        private readonly IConfiguration _config;
+
+        public AuthController(IAuthRepository repo, IConfiguration config)
+        {
+            _repo = repo;
+            _config = config;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserForRegisterDto userForRegisterDto)
+        {
+            //validate request
+            userForRegisterDto.UserName = userForRegisterDto.UserName.ToLower();
+            
+            if (await _repo.UserExists(userForRegisterDto.UserName))
+                return BadRequest("Username already exists");
+            
+            var userToCreate = new Users
+            {
+                Name = userForRegisterDto.UserName
+            };
+
+            var createdUser = await _repo.Register(userToCreate, userForRegisterDto.Password);
+
+            return StatusCode(201);  //will change to createdatroute once we have a place to get data from
+        }
+
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(UserForLoginDto userForLoginDto)
+        {
+            var userFromRepo = await _repo.Login(userForLoginDto.UserName.ToLower(), userForLoginDto.Password);
+
+            if (userFromRepo == null)
+                return Unauthorized();
+
+            //***************  All of this to create a token to send to users logging in *******************
+            
+            //2 claims being sent back as part of token containing user Id and username
+            var claims = new[]
+            {
+                new Claim( ClaimTypes.NameIdentifier, userFromRepo.Id.ToString() ),
+                new Claim( ClaimTypes.Name, userFromRepo.Name)
+            };
+
+            //creating security key
```

Comment:
same, comments aren't really needed, you have the 'new' keyword below which tells me you are creating a new Security Key